### PR TITLE
chore: operator warnings

### LINKED ISSUE #2543: Segments UI: warn users when using the `IN` operator
**Is your feature request related to a problem? Please describe.**

In local evaluation mode, older SDKs will break if they receive an environment containing segments with the `IN` operator. This is clarified here: https://docs.flagsmith.com/clients/overview#in-segment-operator

During segment creation/editing, we should warn the users about this. 

**Describe the solution you'd like**

Whenever the user chooses the `IN` segment operator, display a warning with the above link. The copy should read:

> [Check your SDK version supports the IN operator.](https://docs.flagsmith.com/clients/overview#in-segment-operator)

**Describe alternatives you've considered**

N/A

### PR DESCRIPTION
**Pull request by kyle-ssg**
Thanks for submitting a PR! Please check the boxes below:

- [x] I have run [`pre-commit`](https://docs.flagsmith.com/platform/contributing#pre-commit) to check linting
- [x] I have filled in the "Changes" section below?
- [x] I have filled in the "How did you test this code" section below?
- [x] I have used a [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) title for this Pull Request

## Changes

Fixes:
- Fixed remove button hitbox for segment overrides 

Chore:
- Migrated rule component to typescript
- Added operator placeholders and warnings to config.

Given the following remote config:

```
  {
    "value": "IN",
    "label": "In",
    "warning": "<a href=\"https://docs.flagsmith.com/clients/overview#in-segment-operator\" target=\"_blank\">Check your SDK version supports the IN operator.</a>",
    "valuePlaceholder": "Value1,Value2"
  }
```

Unique warnings will be shown based on the segment's rules:

image

## How did you test this code?

_Please describe._

### CODE REVIEW

File: frontend/package.json [hunk 1]
```diff
@@ -153,7 +153,7 @@
     "nodemon": "^2.0.16",
     "raw-loader": "0.5.1",
     "ssgrtk": "^0.3.4",
-    "testcafe": "^2.6.0",
+    "testcafe": "^3.1.0",
```
Comment (matthewelwell) [inline frontend/package.json]: Is this meant to be in this PR?

**Comment by dabeeeenster** [comment]
This is working perfectly here. 👍

**Review by matthewelwell — APPROVED** [review]
Yep, looks good on the preview. Added one minor comment regarding the testcafe version change.

File: frontend/package.json (hunk 1 above)
Comment (kyle-ssg) [reply frontend/package.json]: Yep

Comment (kyle-ssg) [reply frontend/package.json]: I can split this if we want to review it separately, but really if the tests pass I think it can go through

Comment (matthewelwell) [reply frontend/package.json]: I would rather we keep things like that separate if possible so we have atomic commits, but I'm not _too_ fussy here.

**Review by matthewelwell — APPROVED** [review]

**Pull request merged**